src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,434 @@
+      <_CoreclrResolvedPath Include="@(CrossgenResolvedAssembliesToPublish)"
+                                             Condition="'%(CrossgenResolvedAssembliesToPublish.Filename)'=='libcoreclr'" />
+      <_JitResolvedPath Include="@(CrossgenResolvedAssembliesToPublish)"
+                                             Condition="'%(CrossgenResolvedAssembliesToPublish.Filename)'=='clrjit'" />
+      <_JitResolvedPath Include="@(CrossgenResolvedAssembliesToPublish)"
+                                             Condition="'%(CrossgenResolvedAssembliesToPublish.Filename)'=='libclrjit'" />
+      <_AssembliestoCrossGen Include="$(_RuntimeDir)\**\*.*" />
+    </ItemGroup>
+
+    <NETSdkError Condition="'@(_CoreclrResolvedPath->Count())' &gt; 1"
+    ResourceName="MultipleFilesResolved"
+    FormatArguments="coreclr" />
+
+    <NETSdkError Condition="'@(_CoreclrResolvedPath)'== ''"
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="coreclr" />
+
+    <NETSdkError Condition="'@(_JitResolvedPath->Count())' &gt; 1"
+    ResourceName="MultipleFilesResolved"
+    FormatArguments="jit" />
+
+    <NETSdkError Condition="'@(_JitResolvedPath)'== ''"
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="jit" />
+    
+    
+    
+    <!-- Get the crossgen and jit path-->
+    <PropertyGroup>
+      <_CoreclrPath>@(_CoreclrResolvedPath)</_CoreclrPath>
+      <_JitPath>@(_JitResolvedPath)</_JitPath>
+      <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrPath)))</_CoreclrDir>
+      <_CoreclrPkgDir>$([System.IO.Path]::Combine($(_CoreclrDir),"..\..\..\"))</_CoreclrPkgDir>
+      <_CrossgenDir>$([System.IO.Path]::Combine($(_CoreclrPkgDir),"tools"))</_CrossgenDir>
+      <!-- TODO override with rid specific tools path for x-arch -->
+      <_Crossgen>$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen"))</_Crossgen>
+      <_Crossgen Condition="$(OS.ToUpper().Contains('WINDOWS'))">$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen.exe"))</_Crossgen>
+    </PropertyGroup>
+
+    <NETSdkError Condition="!Exists($(_Crossgen))"
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="$(_Crossgen)" />
+
+    <PropertyGroup>
+      <_Crossgen> $([System.IO.Path]::GetFullPath($(_Crossgen)))</_Crossgen>
+    </PropertyGroup>
+
+    <!-- Create the output directory where crossgened assemblies are going to be present -->
+    <ItemGroup>
+      <_RuntimeOptimizedDirectories Include="@(_AssembliestoCrossGen -> '$(_RuntimeOptimizedDir)\%(RecursiveDir)')"/>
+    </ItemGroup>
+    <MakeDir  Directories="@(_RuntimeOptimizedDirectories)"/>
+
+    <!-- CrossGen the assemblies  -->
+    <Message Text="$(_Crossgen) -readytorun -in %(_AssembliestoCrossGen.Fullpath) -out $([System.IO.Path]::Combine($(_RuntimeOptimizedDir),%(RecursiveDir)%(Filename)%(Extension))) -jitpath $(_JitPath) -platform_assemblies_paths $(_RuntimeRefDir)$(PathSeparator)$(_NetCoreRefDir)"/>
+    <Exec
```

Comment:
http://stackoverflow.com/questions/997047/how-to-run-tasks-in-parallel-in-msbuild

AFAIK, the only way you can parallelize work in MSBuild is to split the work into separate `MSBuild` invocations.  That was the answer 7 years ago, and I'm assuming it is still the same answer.

@rainersigwald @jeffkl @cdmihai - in case they know of a way